File: org/cooklang/Cookware.java
package org.cooklang;

public class Cookware implements DirectionItem {
    public String name;

    private Float quantityFloat;
    private String quantityString;

    public void setQuantityFloat(Float newQuantity) {
        this.quantityFloat = newQuantity;
    }

    public void setQuantityString(String newQuantity) {
        this.quantityString = newQuantity;
    }

    public void setName(String newName) {
        this.name = newName;
    }

    public String toString() {
        return name;
    }
}

File: parser/src/main/java/org/cooklang/parser/Cookware.kt
package org.cooklang.parser

import org.cooklang.parser.DirectionItem

data class Cookware(
    var name: String? = null,
    var quantityFloat: Float? = null,
    var quantityString: String? = null
) : DirectionItem {
    override fun getDisplayString(): String {
        return "${quantityFloat ?: quantityString ?: ""} $name"
    }
}